### PR TITLE
Fix to panic if `ExecStep` has an error when converting to `ExecState`

### PR DESCRIPTION
**Pull request by silathdiir**
Close https://github.com/scroll-tech/zkevm-circuits/issues/66

### CODE REVIEW

File: zkevm-circuits/src/evm_circuit/witness.rs [hunk 1]
```diff
@@ -559,9 +560,41 @@ impl Rw {
 
 impl From<&bus_mapping::circuit_input_builder::ExecStep> for ExecutionState {
     fn from(step: &bus_mapping::circuit_input_builder::ExecStep) -> Self {
-        // TODO: error reporting. (errors are defined in
-        // circuit_input_builder.rs)
-        debug_assert!(step.error.is_none());
+        if let Some(error) = step.error.as_ref() {
+            let state = match error {
+                ExecError::ExecutionReverted => ExecutionState::REVERT,
```
Comment (HAOYUatHZ) [inline zkevm-circuits/src/evm_circuit/witness.rs:565]: `ExecutionState::REVERT`

File: zkevm-circuits/src/evm_circuit/witness.rs [hunk 2]
```diff
@@ -559,9 +560,41 @@ impl Rw {
...
+                ExecError::InvalidOpcode => ExecutionState::ErrorInvalidOpcode,
+                ExecError::StackOverflow => ExecutionState::ErrorStackOverflow,
+                ExecError::StackUnderflow => ExecutionState::ErrorStackUnderflow,
+                ExecError::WriteProtection => ExecutionState::ErrorWriteProtection,
+                ExecError::Depth => ExecutionState::ErrorDepth,
+                ExecError::InsufficientBalance => ExecutionState::ErrorInsufficientBalance,
+                ExecError::ContractAddressCollision => {
+                    ExecutionState::ErrorContractAddressCollision
+                }
+                ExecError::InvalidCode => ExecutionState::ErrorInvalidCreationCode,
```
Comment (HAOYUatHZ) [inline zkevm-circuits/src/evm_circuit/witness.rs:575]: I am not sure whether `ExecError::InvalidCode` is  corresonding to `ExecutionState::ErrorInvalidCreationCode`. 

@lispc @DreamWuGit do you think this line is correct?

File: zkevm-circuits/src/evm_circuit/witness.rs [hunk 3]
```diff
@@ -559,9 +560,41 @@ impl Rw {
...
+                    OogError::ExtCodeCopy => ExecutionState::ErrorOutOfGasEXTCODECOPY,
+                    OogError::ReturnDataCopy => ExecutionState::ErrorOutOfGasRETURNDATACOPY,
+                    OogError::Log => ExecutionState::ErrorOutOfGasLOG,
+                    OogError::Call => ExecutionState::ErrorOutOfGasCALL,
+                    OogError::CallCode => ExecutionState::ErrorOutOfGasCALLCODE,
+                    OogError::DelegateCall => ExecutionState::ErrorOutOfGasDELEGATECALL,
+                    OogError::Create2 => ExecutionState::ErrorOutOfGasCREATE2,
+                    OogError::StaticCall => ExecutionState::ErrorOutOfGasSTATICCALL,
+                },
+            };
```
Comment (HAOYUatHZ) [inline zkevm-circuits/src/evm_circuit/witness.rs:595]: I re-think about it and feel that we'd better have 
```
impl From<ExecError> for ExecutionState
```

to make the codes cleaner

File: zkevm-circuits/src/evm_circuit/witness.rs [hunk 4]
```diff
@@ -557,11 +558,46 @@ impl Rw {
     }
 }
 
+impl From<&ExecError> for ExecutionState {
+    fn from(error: &ExecError) -> Self {
+        match error {
+            ExecError::ExecutionReverted => ExecutionState::REVERT,
```
[the remark this answers is not in the document]
Comment (HAOYUatHZ) [reply zkevm-circuits/src/evm_circuit/witness.rs:564]: sorry I made a typo yesterday. I mean `ExecutionState::ErrorReverted`

**Comment by scroll-dev** [comment]
moved to https://github.com/appliedzkp/zkevm-circuits/pull/333

**Pull request closed without merging**